pyscriptjs/src/python/pyscript/__init__.py

```diff
@@ -709,6 +714,88 @@ def deprecate(name, obj, instead):
     ns["PyScript"] = DeprecatedGlobal("PyScript", PyScript, message)
 
 
+class _PyscriptWebLoop(WebLoop):
+    def __init__(self):
+        orig_loop = asyncio.get_event_loop()
+        super().__init__()
+        asyncio._set_running_loop(orig_loop)
+        self.ready = False
+        self.deferred_handles = []
+
+    def call_later(  # type: ignore[override]
+        self,
+        delay: float,
+        callback: Callable[..., Any],
+        *args: Any,
+        context: contextvars.Context | None = None,
+    ) -> asyncio.Handle:
+        """Arrange for a callback to be called at a given time.
+        Return a Handle: an opaque object with a cancel() method that
+        can be used to cancel the call.
+        The delay can be an int or float, expressed in seconds.  It is
+        always relative to the current time.
+        Each callback will be called exactly once.  If two callbacks
+        are scheduled for exactly the same time, it undefined which
+        will be called first.
+        Any positional arguments after the callback will be passed to
+        the callback when it is called.
+        This uses `setTimeout(callback, delay)`
+        """
+        if delay < 0:
+            raise ValueError("Can't schedule in the past")
+        h = asyncio.Handle(callback, args, self, context=context)
```

Comment:
This is all copied this out of pyodide's webloop (which in turn is copied out of `cpython/Lib/asyncio/base_events.py`). So perhaps rather than tuning it up we should put in a comment explaining that it is copied and where the changes are.